static/js/stream_list.js

```diff
@@ -19,12 +19,48 @@ function active_stream_name() {
     return false;
 }
 
+function filter_streams_by_search(streams) {
+    var search_box = $(".stream-list-filter");
+
+    var search_term = search_box.expectOne().val().trim();
+
+    if (search_term === '') {
+        return streams;
+    }
+
+    var search_terms = search_term.toLowerCase().split(",");
+    search_terms = _.map(search_terms, function (s) {
+        return s.trim();
+    });
+
+    var filtered_streams = _.filter(streams, function (stream) {
+        return _.any(search_terms, function (search_term) {
+            var lower_stream_name = stream.toLowerCase();
+            var idx = lower_stream_name.indexOf(search_term);
```

Comment:
In place of  code(line no 38 to 49), try something like this. 
    
    var lower_stream_name = stream.toLowerCase().split(" ");
    return _.any(lower_stream_name, function (name) {
        return name.indexOf(search_term) === 0;
    });

Can also refer `filter_users_by_search` in activity.js